packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -0,0 +1,251 @@
+import { Construct } from 'constructs';
+import * as cloudfront from '../../aws-cloudfront';
+import { AccessLevel } from '../../aws-cloudfront';
+import * as iam from '../../aws-iam';
+import { IKey } from '../../aws-kms';
+import { IBucket } from '../../aws-s3';
+import { Annotations, Aws, Names, Stack } from '../../core';
+
+const BUCKET_ACTIONS: Record<string, string[]> = {
+  READ: ['s3:GetObject'],
+  WRITE: ['s3:PutObject'],
+  DELETE: ['s3:DeleteObject'],
+};
+
+const KEY_ACTIONS: Record<string, string[]> = {
+  READ: ['kms:Decrypt'],
+  WRITE: ['kms:Encrypt', 'kms:GenerateDataKey*'],
+};
```

Comment:
Should we have have them defined within `origin-access-controls.ts` so later when we add lambda urls we have a single place to track all permissions?

Reply:
Yes I think that's a good idea, will update

Reply:
> Should we have have them defined within `origin-access-controls.ts` so later when we add lambda urls we have a single place to track all permissions?

Two questions:
1. Does the lambda use case involve KMS key? I thought these KMS permissions only applies on the S3 Bucket use case.
2. If we define these in `origin-access-controls.ts`, does that mean we need to export them? If we export them, users can start using them and changing them may lead to breaking changes. Is my understanding correct?

Reply:
1. Setting up OAC for Lambda function url origins does not involve KMS key. 
2. Yes we would export them, I think I see your point about possible breaking changes to external consumers if we change it. But wouldn't it be a breaking change if we didn't export and updated the permissions? Since that would affecting any customers using the construct. Also I see we have permission files that export sets of permissions ([example](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-s3/lib/perms.ts)) which are also used for grant methods.
I will move it back to the s3 origin file though to keep it private and encapsulated, thanks for pointing it out.

Reply:
I think that's a fair callout from Samson. While both scenarios can be a breaking changes, they are technically two different breaking changes. Keeping it in this file and don't allow users to access it directly reduce the chance of breaking changes (if we will ever change the value) for future proofing.

Reply:
That's a good callout. I agree we should keep it private in s3 origin file to minimize breaking changes.